src/main/java/org/snomed/snowstorm/extension/ExtensionAdditionalLanguageRefsetUpgradeService.java

```diff
@@ -147,37 +151,126 @@ private Map<Long, ReferenceSetMember> getReferencedComponents(String branchPath,
 		return result;
 	}
 
-	private List<ReferenceSetMember> addOrUpdateLanguageRefsetComponents(String branchPath, AdditionalRefsetExecutionConfig config, Map<Long, ReferenceSetMember> existing) {
+	private List<ReferenceSetMember> addOrUpdateLanguageRefsetComponents(String branchPath, AdditionalRefsetExecutionConfig config, Map<Long, ReferenceSetMember> languageRefsetMembersToCopy) {
 		BranchCriteria branchCriteria = versionControlHelper.getBranchCriteria(branchPath);
 		List<ReferenceSetMember> result = new ArrayList<>();
 		// batch here as the max allowed in terms query is 65536
-		for (List<Long> batch : partition(existing.keySet(), 10_000)) {
+
+		// Get all concepts that need changing from ToCopy members
+		Set<String> conceptIds = new HashSet<>();
```

Comment:
@QuyenLy87 You can update existing query to return concept id as part of results.